src/sql/parts/profiler/contrib/profilerActions.ts

```diff
@@ -98,11 +101,29 @@ export class ProfilerPause extends Action {
 	}
 
 	public run(input: ProfilerInput): TPromise<boolean> {
-		this.enabled = false;
-		return TPromise.wrap(this._profilerService.pauseSession(input.id).then(() => {
-			input.state.change({ isPaused: true, isStopped: false, isRunning: false });
-			return true;
-		}));
+		if(!this._paused) {
+			return TPromise.wrap(this._profilerService.pauseSession(input.id).then(() => {
```

Comment:
it looks like `pauseSession` is toggling the "pause" state, not really pausing since it could also unpause if I'm reading the code correctly?  Maybe a name like `togglePauseSession` or something like that makes more sense?

Reply:
That definitely makes more sense. Would you suggest changing the name of the action on sqltoolsservice and the sqlops-dataprotocolclient as well, or just changing `_profilerService.pauseSession` into `_profilerService.togglePauseSession`?

Reply:
I would probably change the JSON-RPC method name as well, though that could be done separately at a later time if it introduces complications in this PR.